gammapy/maps/tests/test_base.py

```diff
@@ -22,13 +23,13 @@
 ]
 
 mapbase_args = [
-    (0.1, 10.0, 'wcs', SkyCoord(0.0, 30.0, unit='deg'), None, ''),
-    (0.1, 10.0, 'wcs', SkyCoord(0.0, 30.0, unit='deg'), map_axes[:1], ''),
-    (0.1, 10.0, 'wcs', SkyCoord(0.0, 30.0, unit='deg'), map_axes, 'm^2'),
-    (0.1, 10.0, 'hpx', SkyCoord(0.0, 30.0, unit='deg'), None, ''),
-    (0.1, 10.0, 'hpx', SkyCoord(0.0, 30.0, unit='deg'), map_axes[:1], ''),
-    (0.1, 10.0, 'hpx', SkyCoord(0.0, 30.0, unit='deg'), map_axes, 's^2'),
-    (0.1, 10.0, 'hpx-sparse', SkyCoord(0.0, 30.0, unit='deg'), None, ''),
+    (0.1*u.deg, 10.0*u.deg, 'wcs', SkyCoord(0.0, 30.0, unit='deg'), None, ''),
+    (0.1*u.deg, 10.0*u.deg, 'wcs', SkyCoord(0.0, 30.0, unit='deg'), map_axes[:1], ''),
+    (0.1*u.deg, 10.0*u.deg, 'wcs', SkyCoord(0.0, 30.0, unit='deg'), map_axes, 'm^2'),
+    (0.1*u.deg, 10.0*u.deg, 'hpx', SkyCoord(0.0, 30.0, unit='deg'), None, ''),
+    (0.1*u.deg, 10.0*u.deg, 'hpx', SkyCoord(0.0, 30.0, unit='deg'), map_axes[:1], ''),
+    (0.1*u.deg, 10.0*u.deg, 'hpx', SkyCoord(0.0, 30.0, unit='deg'), map_axes, 's^2'),
+    (0.1*u.deg, 10.0*u.deg, 'hpx-sparse', SkyCoord(0.0, 30.0, unit='deg'), None, ''),
 ]
```

Comment:
I would not  change existing tests here but add some with proper units. You might add one that ensure the call fails with a `UnitConversionError` if the unit is not correct.

Reply:
Ok for the tests. I've modified that piece.
For the 2nd point, I'm not sure I understand. Should we always add a test to check that the units are correct ? If I says width=5*u.km the code will raise an Error anyway saying :
`UnitConversionError: 'km' (length) and 'deg' (angle) are not convertible`